src/components/EditableToDo.vue

```diff
@@ -0,0 +1,206 @@
+<template>
+  <div class="text-center border-2 border-black mx-3.5 rounded-2xl">
+    <ConfirmationModal
+      v-if="isPopUpShown"
+      title="Warning"
+      text="Are you sure you want to delete this item? This operation is permanent
+          and you will not be able to undo this action!"
+      @remove-to-do="removeTodo"
+      @hide-pop-up="hidePopUp"
+    />
+    <div class="flex flex-col m-[18px]">
+      <div class="flex flex-row justify-between items-center">
+        <input
+          v-model="currentTitle"
+          type="text"
+          placeholder="Title"
+          class="w-[197px] font-semibold text-lg lg:text-4xl"
+          :class="currentOpacity"
+          :disabled="isDropDownShown"
+        />
+        <div class="flex flex-row space-x-[5px] items- lg:hidden items-center">
+          <div
+            v-for="(priority, ind) in priorityList"
+            class="p-1 rounded-full h-fit"
+            :class="getColorOfPriority(priority), styleOfPriorityButton(ind)"
+            @click="changePriority(priority)"
+          ></div>
+        </div>
+        <div class="hidden lg:flex flex-col" :class="positionOfDropDownButton">
+          <div
+            class="text-lg text-white font-semibold rounded-full"
+            :class="dropDownStyle"
+          >
+            <PrimaryButton
+              class="ml-5 px-0 py-0 mr-3 flex items-center"
+              :additional-classes="textColorOfDropDown"
```

Comment:
You don't need to have `additional-classes` prop in your PrimaryButton component you can add `textColorOfDropDown` as dynamic class directly to it